core/src/main/java/hudson/security/SidACL.java

```diff
@@ -147,4 +161,9 @@ protected Boolean hasPermission(Sid p, Permission permission) {
     }
 
     private static final Logger LOGGER = Logger.getLogger(SidACL.class.getName());
+    private final Cache<Authentication, PrincipalSid> principalSidCache = CacheBuilder.newBuilder()
+            .softValues()
```

Comment:
Are `softValues()` needed for the cache? From https://github.com/google/guava/wiki/CachesExplained#reference-based-eviction:

> CacheBuilder.softValues() wraps values in soft references. Softly referenced objects are garbage-collected in a globally least-recently-used manner, in response to memory demand. Because of the performance implications of using soft references, we generally recommend using the more predictable maximum cache size instead. Use of softValues() will cause values to be compared using identity (==) equality instead of equals().